tests/js/object-tests.js

```diff
@@ -777,4 +777,50 @@ module.exports = {
       obj.getPropertyType("foo");
     }, new Error("No such property: foo"));
   },
+
+  testObjectSpread: function () {
+    let realm = new Realm({ schema: [schemas.IntPrimary] });
```

Comment:
We need to test this with a class that extends `Realm.Object`, I believe this has a bug.